backend/Squirrel.Core/Squirrel.Core.BLL/Services/MongoService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.Extensions.Options;
+using MongoDB.Driver;
+using Squirrel.Core.BLL.Interfaces;
+using Squirrel.Core.Common.Models;
+using Squirrel.Core.DAL.Entities;
+
+namespace Squirrel.Core.BLL.Services
+{
+    public class MongoService<T> : IMongoService<T> where T : Entity<long>
+    {
+        private readonly IMongoCollection<T> _mongoCollection;
+
+        public MongoService(IOptions<MongoDatabaseConnectionSettings> mongoDatabaseSettings, string collectionName)
+        {
+            var mongoClient = new MongoClient(mongoDatabaseSettings.Value.ConnectionString);
+
+            var mongoDatabase = mongoClient.GetDatabase(mongoDatabaseSettings.Value.DatabaseName);
+
+            _mongoCollection = mongoDatabase.GetCollection<T>(collectionName);
+        }
+
+        public async Task<T> CreateAsync(T entity)
+        {
+            await _mongoCollection.InsertOneAsync(entity);
```

Comment:
probably

vare created = await _mongoCollection.InsertOneAsync(entity);

return created ;

Reply:
this method doesn't return entity